src/routes/v1/transit.ts

```diff
@@ -93,6 +94,50 @@ export type PostTransitRequestBody = RawSearchParams & {
     cursor?: string
 }
 
+async function recursiveTransitSearch(
+    params: SearchParams,
+    extraHeaders: ExtraHeaders,
+    attempt = 0,
+): Promise<{
+    tripPatterns: TripPatternParsed[]
+    queries: GraphqlQuery[]
+    customCursor: string | undefined
+}> {
+    const maxAttempts = 10
+
+    const {
+        tripPatterns,
+        queries,
+        previousPageCursor,
+        nextPageCursor,
+        routingErrors,
+    } = await searchTransit(params, extraHeaders)
+
+    const hasNoTransitError = routingErrors?.some(
+        (error) =>
+            error.code === RoutingErrorCode.NoTransitConnectionInSearchWindow,
+    )
+
+    const otpCursor = params.arriveBy ? previousPageCursor : nextPageCursor
+
+    let customCursor
+    if (otpCursor) {
+        customCursor = generateCursor(params, otpCursor)
+    }
+
+    if (hasNoTransitError && attempt < maxAttempts) {
+        const updatedParams = parseCursor(customCursor)?.params || params
```

Comment:
Man trenger kanskje ikke fortsette om man ikke klarer lage oppdaterte params?

```suggestion
    const updatedParams = parseCursor(customCursor)?.params
    if (hasNoTransitError && attempt < maxAttempts && updatedParams) {
```

Reply:
Agrees